src/lib.rs

```diff
@@ -122,15 +125,32 @@ mod platform {
             let task = task_for_pid(self.pid);
             if task.is_err() { return task.map(|_| ()) }
 
-            unsafe {
-                let mut read: mach_msg_type_number_t = 0;
+            let page_addr      = (addr as i64 & (-4096)) as mach_vm_address_t;
+	        let last_page_addr = ((addr as i64 + buf.len() as i64 + 4095) & (-4096)) as mach_vm_address_t;
+            let page_size      = last_page_addr as usize - page_addr as usize;
```

Comment:
Yeah I have a bunch of those [opensource.apple.com/source/xnu](https://opensource.apple.com/source/xnu) links bookmarked. (So fortunate for those FOSS licenses ensuring these header & source files are public.) Will put together a comment documenting how all these pieces (I think) fit together. Will also make sure that `vm_deallocate` gets called once we copy the buffer over into the program's memory.